test/with_dummyserver/test_no_ssl.py

```diff
@@ -21,13 +21,22 @@ def test_simple(self) -> None:
             assert r.status == 200, r.data
 
 
-class TestHTTPSWithoutSSL(HTTPSDummyServerTestCase, TestWithoutSSL):
+class TestHTTPSWithoutSSL(HTTPDummyProxyTestCase, TestWithoutSSL):
     def test_simple(self) -> None:
         with urllib3.HTTPSConnectionPool(
-            self.host, self.port, cert_reqs="NONE"
+            self.https_host, self.https_port, cert_reqs="NONE"
         ) as pool:
             with pytest.warns(InsecureRequestWarning):
                 try:
                     pool.request("GET", "/")
                 except urllib3.exceptions.SSLError as e:
                     assert "SSL module is not available" in str(e)
+
+    def test_simple_proxy(self) -> None:
```

Comment:
It would be good to test both HTTPS->HTTP and HTTPS->HTTPS proxy scenarios.

Reply:
What is the expected HTTPS -> HTTP proxy scenario? Currently, it throws an InsecureRequestWarning as `conn.is_verified` is set to `False` while using HTTPs->HTTP connection. Should it only throw `InsecureProxyWarning` ?

Reply:
If I understand correctly, this Seth's suggestion is related to https://github.com/urllib3/urllib3/pull/3070#discussion_r1232951507.
You change the message for `InsecureRequestWarning`, it would be good to test what host is included in the message when a proxy is used as well as changing `test_simple` to check what host is used there.